Source/Core/Common/MsgHandler.h

```diff
@@ -30,22 +31,35 @@ void RegisterStringTranslator(StringTranslator translator);
 
 std::string GetStringT(const char* string);
 
-bool MsgAlert(bool yes_no, MsgType style, const char* format, ...)
-#ifdef __GNUC__
-    __attribute__((format(printf, 3, 4)))
-#endif
-    ;
-
-bool MsgAlertFmtImpl(bool yes_no, MsgType style, fmt::string_view format,
-                     const fmt::format_args& args);
+bool MsgAlertFmtImpl(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
+                     int line, fmt::string_view format, const fmt::format_args& args);
 
 template <std::size_t NumFields, typename S, typename... Args>
-bool MsgAlertFmt(bool yes_no, MsgType style, const S& format, const Args&... args)
+bool MsgAlertFmt(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
+                 int line, const S& format, const Args&... args)
 {
   static_assert(NumFields == sizeof...(args),
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
-  return MsgAlertFmtImpl(yes_no, style, format, fmt::make_args_checked<Args...>(format, args...));
+  return MsgAlertFmtImpl(yes_no, style, log_type, file, line, format,
+                         fmt::make_args_checked<Args...>(format, args...));
+}
+
+template <std::size_t NumFields, bool has_non_positional_args, typename S, typename... Args>
+bool MsgAlertFmtT(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
+                  int line, const S& format, fmt::string_view translated_format,
+                  const Args&... args)
+{
+  static_assert(!has_non_positional_args,
+                "Translatable strings must use positional arguments (e.g. {0} instead of {})");
+  static_assert(NumFields == sizeof...(args),
+                "Unexpected number of replacement fields in format string; did you pass too few or "
+                "too many arguments?");
+  // This is purely to verify that things work in the base language (English).
+  // We can't compile-time check translations unfortunately.
+  fmt::make_args_checked<Args...>(format, args...);
```

Comment:
Is this guaranteed to be called, vs. a compiler simply omitting it? Should we explicitly mark it as ignored/discarded?

Reply:
We're calling it purely for the static assertions it has; the result doesn't matter (and in fact, it would be good if the compiler optimized out the runtime-related aspects; it probably does).

Reply:
Yeah, but does this cause the static assert to trigger even when it is optimized out? If so, good, and you can ignore this one :D

Reply:
I would be very annoyed if compilers could look at a function and decide that because it has no side-effects at runtime, its `static_assert`s can be ignored.  That's not to say that's not how it works, just that it would be very annoying if that was how it works.

But on further investigation, we can use the arg-list from the English string as a parameter for the non-English string; both `make_format_args` and `make_args_checked` return `format_arg_store`, and `make_args_checked`'s return value doesn't depend on the format string (only the checking logic), so the return value doesn't need to go unused.